src/coord/src/catalog/storage.rs

```diff
@@ -220,6 +222,38 @@ impl Connection {
         res
     }
 
+    /// Sets catalog's `cluster_id` setting on initialization or gets that value.
+    fn set_or_get_cluster_id(sqlite: &mut rusqlite::Connection) -> Result<Uuid, Error> {
+        let tx = sqlite.transaction()?;
+        let current_setting: Option<String> = tx
+            .query_row(
+                "SELECT value FROM settings WHERE name = 'cluster_id';",
+                params![],
+                |row| row.get(0),
+            )
+            .optional()?;
+
+        let res = match current_setting {
+            // Server init
+            None => {
+                // Generate a new version 4 UUID. These are generated from random input.
+                let cluster_id = Uuid::new_v4();
+                tx.execute(
+                    "INSERT INTO settings VALUES ('cluster_id', ?);",
+                    params![cluster_id.to_string()],
+                )?;
+                Ok(cluster_id)
+            }
+            // Server reboot
+            Some(cs) => {
+                let current_setting = Uuid::parse_str(&cs).unwrap();
```

Comment:
Can you propagate the error here rather than unwrapping? 

Alternatively you could use `SqlVal` to handle the serialization automatically, which will also push the error handling into the calls to `query_row` or `execute`:

```diff
diff --git a/src/coord/src/catalog/storage.rs b/src/coord/src/catalog/storage.rs
index f4fcad3a4..e7148ff4b 100644
--- a/src/coord/src/catalog/storage.rs
+++ b/src/coord/src/catalog/storage.rs
@@ -225,7 +225,7 @@ impl Connection {
     /// Sets catalog's `cluster_id` setting on initialization or gets that value.
     fn set_or_get_cluster_id(sqlite: &mut rusqlite::Connection) -> Result<Uuid, Error> {
         let tx = sqlite.transaction()?;
-        let current_setting: Option<String> = tx
+        let current_setting: Option<SqlVal<Uuid>> = tx
             .query_row(
                 "SELECT value FROM settings WHERE name = 'cluster_id';",
                 params![],
@@ -240,15 +240,12 @@ impl Connection {
                 let cluster_id = Uuid::new_v4();
                 tx.execute(
                     "INSERT INTO settings VALUES ('cluster_id', ?);",
-                    params![cluster_id.to_string()],
+                    params![SqlVal(cluster_id)],
                 )?;
                 Ok(cluster_id)
             }
             // Server reboot
-            Some(cs) => {
-                let current_setting = Uuid::parse_str(&cs).unwrap();
-                Ok(current_setting)
-            }
+            Some(cs) => Ok(cs.0),
         };
         tx.commit()?;
         res
```